src/app/components/shared/timezone/timezone.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, Input, OnChanges } from "@angular/core";
+import { DateTimeTimezone } from "@interfaces/apiInterfaces";
+import {
+  AbstractModel,
+  isUnresolvedModel,
+  UnresolvedModel,
+} from "@models/AbstractModel";
+import { Site } from "@models/Site";
+
+@Component({
+  selector: "baw-timezone",
+  template: `
+    <span [ngbTooltip]="timezone">
+      {{ formattedDate }}
+    </span>
+  `,
+  styles: [
+    `
+      span {
+        text-decoration: underline dotted;
+        cursor: help;
+        text-decoration-skip-ink: none;
+      }
+    `,
+  ],
+})
+export class TimezoneComponent implements OnChanges {
+  @Input() public dateTime: DateTimeTimezone;
+  @Input() public site: UnresolvedModel | Site;
+  @Input() public dateFormat: (date: DateTimeTimezone) => string = (date) =>
```

Comment:
Good escape hatch thinking.

I'm wondering though if we can just accept one of a few predefined types? e.g. `datetime`, `date`, `time`.
That way we'd be sure times are only shown in our formats, but we can still change format for suitability to the current use of the component